DevOpsCLI/Commands/Release/ReleaseCreateCommand.cs

```diff
@@ -67,6 +74,27 @@ protected override int OnExecute(CommandLineApplication app)
                 Reason = this.Reason,
             };
 
+            if (!string.IsNullOrEmpty(this.Artifacts))
+            {
+                dynamic artifacts = JArray.Parse(this.Artifacts);
+
+                List<ReleaseArtifactMetadata> releaseArtifactMetadataList = new List<ReleaseArtifactMetadata>();
+                foreach (var artifact in artifacts)
+                {
+                    releaseArtifactMetadataList.Add(
+                        new ReleaseArtifactMetadata
+                        {
+                            Alias = artifact.Alias,
+                            InstanceReference = new BuildVersion
+                            {
+                                Id = artifact.Id,
+                            },
+                        });
+                }
```

Comment:
Any validation here that those variables will in fact exist in the artifact item?

Maybe you can create a DTO class the deserialize that.

```csharp
JsonConvert.DeserializeObject<List<ArtifactDTO>>(this.Artifacts);
```

And then be sure that the required values are coming from the json string otherwise throw an exception indicating the error.

Reply:
The code was refactored to validate object resulting from json parameter